x/ccv/provider/keeper/msg_server.go

```diff
@@ -357,58 +367,59 @@ func (k msgServer) CreateConsumer(goCtx context.Context, msg *types.MsgCreateCon
 }
 
 // UpdateConsumer updates the record of a consumer chain
 func (k msgServer) UpdateConsumer(goCtx context.Context, msg *types.MsgUpdateConsumer) (*types.MsgUpdateConsumerResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	resp := types.MsgUpdateConsumerResponse{}
+
 	consumerId := msg.ConsumerId
 
-	phase := k.Keeper.GetConsumerPhase(ctx, consumerId)
-	if phase != types.ConsumerPhase_CONSUMER_PHASE_REGISTERED &&
-		phase != types.ConsumerPhase_CONSUMER_PHASE_INITIALIZED &&
-		phase != types.ConsumerPhase_CONSUMER_PHASE_LAUNCHED {
-		return &types.MsgUpdateConsumerResponse{}, errorsmod.Wrapf(types.ErrInvalidPhase,
+	if !k.Keeper.IsConsumerActive(ctx, consumerId) {
+		return &resp, errorsmod.Wrapf(types.ErrInvalidPhase,
 			"cannot update consumer chain that is not in the registered, initialized, or launched phase: %s", consumerId)
 	}
 
 	ownerAddress, err := k.Keeper.GetConsumerOwnerAddress(ctx, consumerId)
 	if err != nil {
-		return &types.MsgUpdateConsumerResponse{}, errorsmod.Wrapf(types.ErrNoOwnerAddress, "cannot retrieve owner address %s", ownerAddress)
+		return &resp, errorsmod.Wrapf(types.ErrNoOwnerAddress, "cannot retrieve owner address %s", ownerAddress)
 	}
 
 	if msg.Signer != ownerAddress {
-		return &types.MsgUpdateConsumerResponse{}, errorsmod.Wrapf(types.ErrUnauthorized, "expected owner address %s, got %s", ownerAddress, msg.Signer)
+		return &resp, errorsmod.Wrapf(types.ErrUnauthorized, "expected owner address %s, got %s", ownerAddress, msg.Signer)
 	}
 
 	// The new owner address can be empty, in which case the consumer chain does not change its owner.
 	// However, if the new owner address is not empty, we verify that it's a valid account address.
 	if strings.TrimSpace(msg.NewOwnerAddress) != "" {
 		if _, err := k.accountKeeper.AddressCodec().StringToBytes(msg.NewOwnerAddress); err != nil {
-			return &types.MsgUpdateConsumerResponse{}, errorsmod.Wrapf(types.ErrInvalidNewOwnerAddress, "invalid new owner address %s", msg.NewOwnerAddress)
+			return &resp, errorsmod.Wrapf(types.ErrInvalidNewOwnerAddress, "invalid new owner address %s", msg.NewOwnerAddress)
 		}
 
 		k.Keeper.SetConsumerOwnerAddress(ctx, consumerId, msg.NewOwnerAddress)
 	}
 
 	if msg.Metadata != nil {
 		if err := k.Keeper.SetConsumerMetadata(ctx, consumerId, *msg.Metadata); err != nil {
-			return &types.MsgUpdateConsumerResponse{}, errorsmod.Wrapf(types.ErrInvalidConsumerMetadata,
+			return &resp, errorsmod.Wrapf(types.ErrInvalidConsumerMetadata,
 				"cannot set consumer metadata: %s", err.Error())
 		}
 	}
 
 	// get the previous spawn time so that we can use it in `PrepareConsumerForLaunch`
-	var previousSpawnTime time.Time
-	previousInitializationParameters, err := k.Keeper.GetConsumerInitializationParameters(ctx, msg.ConsumerId)
-	if err == nil {
-		previousSpawnTime = previousInitializationParameters.SpawnTime
+	previousInitializationParameters, err := k.Keeper.GetConsumerInitializationParameters(ctx, consumerId)
+	if err != nil {
```

Comment:
I think the first time the initialization parameters are set, this would return `err != nil` and hence we would fail although we should not.

Reply:
When creating a consumer (via CreateConsumer), all the params are initialized even if not provided. So if this get is returning error, then there is an issue with the state.